### PR TITLE
Rewrite RedundantConstructorCheck without PMD

### LINKED ISSUE #191: `RedundantConstructorCheck` does not detect records
### Summary

Having a default constructor that replicates the constructor generated by a record, does not trigger the lint.

Additionally there is extra syntax: `MyRecord { ... }` for initalizing? If this is empty, it could be removed?

### Lint Name

REDUNDANT_DEFAULT_CONSTRUCTOR

### Reproducer

```java
record MyRecord(String value, int number) {
    MyRecord { // Not Ok?
    }

    MyRecord(String value, int number) { // Not OK
        this.value = value;
        this.number = number;
    }
}
```

### PR DESCRIPTION
**Pull request by fschwalbe**
Closes #191

### CODE REVIEW

File: autograder-core/src/main/java/de/firemage/autograder/core/check/complexity/RedundantConstructorCheck.java [hunk 1]
```diff
@@ -2,14 +2,62 @@
...
+                    && isEmpty(ctor.getBody())
+                    && ctor.getThrownTypes().isEmpty();
+                if (isRedundant) {
+                    addLocalProblem(ctor, MESSAGE, ProblemType.REDUNDANT_DEFAULT_CONSTRUCTOR);
+                }
+            }
+        });
+    }
+
+    private boolean hasDefaultVisibility(CtClass<?> type, CtConstructor<?> ctor) {
```
Comment (Luro02) [inline autograder-core/src/main/java/de/firemage/autograder/core/check/complexity/RedundantConstructorCheck.java:44]: Why default visibility? The java generated default constructor is public

Comment (fschwalbe) [reply autograder-core/src/main/java/de/firemage/autograder/core/check/complexity/RedundantConstructorCheck.java:64]: https://docs.oracle.com/javase/specs/jls/se22/html/jls-8.html#jls-8.8.9

> The default constructor has the same access modifier as the class, unless the class lacks an access modifier, in which case the default constructor has package access

File: autograder-core/src/test/java/de/firemage/autograder/core/check/complexity/TestRedundantConstructor.java [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+            """
+                public class Test {
+                    public Test() {
+                    
+                    }
+                }
+                """
+        );
+    }
+
```
Comment (Luro02) [inline autograder-core/src/test/java/de/firemage/autograder/core/check/complexity/TestRedundantConstructor.java:35]: Another thing to test:
```java
public class Test {
    private static class InnerClass {
        private InnerClass() {
        }
    }
}
```

**Review by Luro02 — CHANGES_REQUESTED** [review]
Other than that, looks good to me. I will merge this once I can introduce breaking changes.

**Review by Luro02 — APPROVED** [review]

**Comment by Luro02** [comment]
Thank you for the pull request, really appreciate it :)

**Pull request merged**